src/unix/unix.c

```diff
@@ -232,18 +232,26 @@ define_closure_function(1, 1, context, unix_fault_handler,
     } else if (is_page_fault(ctx->frame)) {
         pf_debug("page fault, vaddr 0x%lx, ctx %p, type %d, pc 0x%lx",
                  vaddr, ctx, ctx->type, ctx->frame[SYSCALL_FRAME_PC]);
+        if (vaddr < PAGESIZE /* no zero page access */ ||
+            vaddr > (USER_LIMIT - PAGESIZE)) {
```

Comment:
Is there something special about the last user page that it can't be used?

Reply:
Thanks, this shouldn't be calling bug. I'll change this to segv for zero page or kernel memory access. I also mistook vaddr as being the page address, so the latter expression should be ```vaddr >= USER_LIMIT```.